components/tab/src/tab-bar/scroll-bar.js

```diff
@@ -1,27 +1,70 @@
 import propTypes from '@dhis2/prop-types'
-import { colors } from '@dhis2/ui-constants'
 import { IconChevronRight16, IconChevronLeft16 } from '@dhis2/ui-icons'
 import cx from 'classnames'
 import React, { Component, createRef } from 'react'
 import { animatedSideScroll } from './animated-side-scroll.js'
 import { detectHorizontalScrollbarHeight } from './detect-horizontal-scrollbar-height.js'
+import { ScrollButton } from './scroll-button.js'
 
 class ScrollBar extends Component {
     scrollBox = createRef()
     scrollArea = createRef()
-    state = {
-        scrolledToStart: false,
-        scrolledToEnd: false,
-    }
     horizontalScrollBarHeight = detectHorizontalScrollbarHeight()
+    scrollBoxResizeObserver = null
+
+    constructor(props) {
+        super(props)
+        this.state = {
+            scrolledToStart: true,
+            scrolledToEnd: false,
+            // used to initially hide the entire content to prevent flickering
+            hideScrollButtonsInitialized: false,
+            // hide buttons initially to simplify calculations
+            hideScrollButtons: false,
+        }
+    }
```

Comment:
```suggestion
    constructor(props) {
        super(props)
        this.state = {
            scrolledToStart: true,
            scrolledToEnd: false,
            // used to initially hide the entire content to prevent flickering
            hideScrollButtonsInitialized: false,
            // hide buttons initially to simplify calculations
            hideScrollButtons: false,
        }
        this.scrollBoxResizeObserver = new ResizeObserver(this.calculateShouldHideButtons)
    }
```